File: java/mlt-cli/src/main/java/org/maplibre/mlt/cli/SerialTaskRunner.java
package org.maplibre.mlt.cli;

import jakarta.annotation.Nullable;

public class SerialTaskRunner implements TaskRunner {
  public SerialTaskRunner() {}

  public int getThreadCount() {
    return 0;
  }

  public void run(@Nullable Runnable task) {
    if (task != null) {
      task.run();
    }
  }

  public void awaitTermination() throws InterruptedException {}

  public void shutdown() {}
}

File: java/mlt-cli/src/main/kotlin/org/maplibre/mlt/cli/SerialTaskRunner.kt
package org.maplibre.mlt.cli

class SerialTaskRunner : TaskRunner {
    override val threadCount: Int
        get() = 0

    override fun run(task: Runnable?) {
        if (task != null) {
            task.run()
        }
    }

    override fun awaitTermination() {
    }

    override fun shutdown() {}
}
